src/main/webapp/script.js

```diff
@@ -260,3 +280,137 @@ function checkPercentage(headerColumn) {
 }
+
+// Returns an object containing all of the relevant data in order
+// to render the geoJson map of the counties.
+function getMapsData(censusDataArray) {
+  const countyToPopMap = new Map();
+  const populationsList = [];
+  // Get rid of the header
+  const censusArray = censusDataArray.slice(1);
+  Array.prototype.forEach.call(censusArray, (county) => {
+    // The current county strings are in a layout like this:
+    // "Contra Costa County, California"
+    // and we need to get them like this "Contra Costa"
+    const countyAndStateArray = county[0].split(',');
+    // ^^ ["Contra Costa County", "California"]
+    const countyArray = countyAndStateArray[0].split(' ');
+    // ^^ ["Contra", "Costa", "County"]
+    let countyString = '';
+    let i;
+    // Get all strings except for the last one
+    for (i = 0; i < countyArray.length - 1; i++) {
+      countyString += countyArray[i];
+      if (i !== countyArray.length - 2) {
+        countyString += ' ';
+      }
+    }
+    // Map the population to the county
+    countyToPopMap[countyString] = county[1];
+    populationsList.push(parseInt(county[1]));
+    });
+  const minAndMax = getMinAndMaxPopulation(populationsList);
+  return {map: countyToPopMap,
+    minValue: minAndMax.min, maxValue: minAndMax.max};
+}
+
+// Returns an object with the min and max population of the
+// populations returned by the census API.
+function getMinAndMaxPopulation(populationArray) {
+  let max = populationArray[0];
+  let min = 0;
+  let i;
+  for (i = 1; i < populationArray.length; i++) {
+    if (populationArray[i] > max) {
+      max = populationArray[i];
+    } else if (populationArray[i] < min) {
+      min = populationArray[i];
+    }
+  }
+  return {max: max, min: min};
+}
+
+// Takes in mapsData object which has a data structure that maps
+// counties to populations, a max population, and a min population.
+// Initializes the geoJson and adds multiple event listeners.
+function displayCountyGeoJson(mapsData, stateName) {
+  const map = new google.maps.Map(document.getElementById('map'), {
+    zoom: stateInfo[stateName].zoomLevel,
+    center: {lat: stateInfo[stateName].lat, lng: stateInfo[stateName].lng},
+  });
+
+  const countyToPopMap = mapsData.map;
+  const maxPopulation = mapsData.maxValue;
+  const minPopulation = mapsData.minValue;
+  const f = chroma.scale(['white', 'blue']).domain([minPopulation,
+    maxPopulation]);
+  const geoData = getGeoData(stateName, true);
+
+  map.data.addGeoJson(geoData);
+  map.data.forEach(function(feature) {
+    map.data.setStyle((feature) => {
+      return {
+        fillColor: f(countyToPopMap[feature.j.name]).toString(),
+      };
+    });
+  });
+
+  const openInfoWindows = [];
+  map.data.addListener('mouseover', function(event) {
+    map.data.overrideStyle(event.feature, {
+      fillColor: '#00ffff',
+    });
+    const contentString = '<p>' + event.feature.j.name +
+    '<p>Population: ' + countyToPopMap[event.feature.j.name];
```

Comment:
We will want to make the "population" descriptor more flexible here depending on what is being displayed, but that can be something for later